presto-hive/src/main/java/com/facebook/presto/hive/DirectoryLister.java

```diff
@@ -16,12 +16,15 @@
 import org.apache.hadoop.fs.FileSystem;
 import org.apache.hadoop.fs.LocatedFileStatus;
 import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.PathFilter;
 
 import java.util.Iterator;
+import java.util.Optional;
 
 import static com.facebook.presto.hive.util.HiveFileIterator.NestedDirectoryPolicy;
 
 public interface DirectoryLister
 {
     Iterator<LocatedFileStatus> list(FileSystem fs, Path path, NamenodeStats namenodeStats, NestedDirectoryPolicy nestedDirectoryPolicy);
+    Iterator<LocatedFileStatus> list(FileSystem fs, Path path, NamenodeStats namenodeStats, NestedDirectoryPolicy nestedDirectoryPolicy, Optional<PathFilter> pathFilter);
```

Comment:
nit: extra line between methods.

I think we could just use the same interface and default other use cases to be `Optional.empty()`, but I also realized that this might break other private connectors. @wenleix , @arhimondr do you have a suggestion here?